### ISSUE TITLE
Daemon support

### ISSUE THREAD

**Issue by buger** [buger/gor#180]
Currently Gor do not support demonizing natively, there is lot room for improvement.

/cc #179

**Comment by jakubholynet** [comment]
It is worth mentioning (for those who stumble upon this issue) that Gor can be daemonized easily via http://www.libslack.org/daemon/

**Comment by buger** [comment]
I like Sidekiq approach and agree with this https://www.mikeperham.com/2015/07/16/sidekiq-and-upstart/

* We don’t need to daemonize. Modern daemons should never daemonize themselves.
* We don’t need PID files. PID files are legacy from years ago and their use should signal that something is wrong.
* We don’t need to specify our own log files. App will output to stdout; Upstart will direct stdout to a file within /var/log/upstart/ and automatically rotate those log files for you - no logrotate setup necessary!

In other words, stop reinventing the wheel and let the operating system do the hard work for you! I hope this makes your Sidekiq deployment cleaner and more stable.
